sdk/python/foundation-models/system/content-safety/deploy_with_content_safety.ipynb

```diff
@@ -0,0 +1,803 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "llama_endpoint = ml_client.online_endpoints.get(name=llama_endpoint_name)\n",
+    "llama_endpoint.tags = {\"AllowlistedObjectIds\": uai_principal_id}\n",
+    "ml_client.online_endpoints.begin_create_or_update(llama_endpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 8.2 Get an AuthorizationManagementClient to list Role Definitions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.mgmt.authorization import AuthorizationManagementClient\n",
+    "from azure.mgmt.authorization.v2018_01_01_preview.models import RoleDefinition\n",
+    "import uuid\n",
+    "\n",
+    "role_definition_client = AuthorizationManagementClient(\n",
+    "    credential=credential,\n",
+    "    subscription_id=subscription_id,\n",
+    "    api_version=\"2018-01-01-preview\",\n",
+    ")\n",
+    "\n",
+    "from azure.mgmt.authorization.v2020_10_01_preview.models import RoleAssignmentCreateParameters\n",
+    "\n",
+    "role_assignment_client = AuthorizationManagementClient(\n",
+    "    credential=credential,\n",
+    "    subscription_id=subscription_id,\n",
+    "    api_version=\"2020-10-01-preview\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 8.3 Grant the user identity access to the Content Safety\n",
+    "Cognitive Services User role is required to access the Content Safety."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role_name = \"Cognitive Services User\"\n",
+    "scope = aacs_resource_id\n",
+    "\n",
+    "role_defs = role_definition_client.role_definitions.list(scope=scope)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "role_assignment_client.role_assignments.create(\n",
+    "    scope=scope,\n",
+    "    role_assignment_name=str(uuid.uuid4()),\n",
+    "    parameters=RoleAssignmentCreateParameters(\n",
+    "        role_definition_id=role_def.id,\n",
+    "        principal_id=uai_principal_id,\n",
+    "        principal_type=\"ServicePrincipal\",\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 8.4 Assign AcrPull at the workspace container registry scope\n",
+    "Since we will create the safety enabled llama endpoint with User Assigned Identity, the user's managed identity must have Storage blob data reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
```

Comment:
```suggestion
    "Since we will create the content safety enabled LlaMa endpoint with User Assigned Identity, the user's managed identity must have Storage blob data reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
```

Reply:
fixed.